engine.py

```diff
@@ -1,7 +1,9 @@
 class Registry:
-    instances : dict = dict()
-    declarations : dict = dict()
-    visited : set = set()
+
+    def __init__(self):
+        self.instances : dict = dict()
```

Comment:
no need to decorate type in this case, it is evident due to the attribution. It was decorated before because of style lints.

Reply:
ye, that didn't make much sense now that I think about it